src/api.py

```diff
@@ -0,0 +1,28 @@
+import requests 
+import hashlib
+
+
+class response:
+    def __init__(self, filename, type_of_data, size, url, checksum):
+        self.filename = filename
+        self.type_of_data = type_of_data
+        self.size=size
+        self.url=url
+        self.checksum=checksum
+
+
+def fetch(url):
+    try:
+        con = requests.head(url)
+        r = requests.get(url)
+        for i in range(len(url.split('/'))-1):
```

Comment:
What is this code section doing? it is rather complex and hard to read? Can you make this a function of its own with a docstring and tests?